clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalTable.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.client;
+
+import com.datastrato.gravitino.Audit;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.dto.rel.TableDTO;
+import com.datastrato.gravitino.dto.responses.EntityListResponse;
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.exceptions.PartitionAlreadyExistsException;
+import com.datastrato.gravitino.rel.Column;
+import com.datastrato.gravitino.rel.SupportsPartitions;
+import com.datastrato.gravitino.rel.Table;
+import com.datastrato.gravitino.rel.expressions.distributions.Distribution;
+import com.datastrato.gravitino.rel.expressions.sorts.SortOrder;
+import com.datastrato.gravitino.rel.expressions.transforms.Transform;
+import com.datastrato.gravitino.rel.partitions.Partition;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+public class RelationalTable implements Table, SupportsPartitions {
```

Comment:
You should also add the UTs to cover this class.

Reply:
Will do it in another PR

Reply:
added in this PR now